transform/mattermost-analytics/models/marts/sales/_sales__models.yml

```diff
@@ -0,0 +1,17 @@
+version: 2
+
+
+models:
+  - name: fct_subscription_history
```

Comment:
If I understand correctly, the intention is to join these data with stripe data. Perhaps the join can happen here so that data become available in a form that's easier to handle in Looker.

Reply:
Do you have an example in mind of what this might look like? I don't see another fct model in the marts folder for Stripe - I also want to note that this may not only join against Subscription data, but with customer data, invoice data, etc - we don't actually know what'll be necessary in the future. I'm not sure the ramifications of leaving it vague or being more specific here - is it mostly performance?

Reply:
Mostly performance and complexity. Adding complex join logic in looker can cause issues in both. The idea is that the models in the mart layer are following the star schema conventions (or similar). Short story is that a fact table contains either numeric/boolean values or keys to dimension tables. Dimension tables contain keys and anything that can be used to filter the fact table by. There's a ton of details here that depend on the use case. 

Since it's not possible to predict what's coming in the future, we can start with the bare minimum for the current use case. Future PRs can extend existing fact/dimension tables or introduce new ones as needed.

Reply:
I think that's a good plan - once I've really narrowed down what the dashboard will look like I can definitely open a subsequent PR to address perf concerns. Thanks!